cli/cmd/testdata/inject_gettest_deployment.good.golden.yml

```diff
@@ -42,6 +42,8 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: status.podIPs
+        - name: LINKERD2_PROXY_INBOUND_PORTS
+          value: 9090,9090
```

Comment:
It's not a major issue, but we could probably de-dup this list so that we don't have the same port multiple times in the list.

To be safe we may also want to make sure these values are wrapped by quotations. It looks like sometimes they are and sometimes they are not in this change. With how YAML can sometimes render `0` or `1` as truthy that could lead to an error here—that is extremely edge case though.

Reply:
I looked deeper into this. Yamls like this one that don't have quotes are the result of applying a json patch during injection. In this particular case the patch looks something like this:
```json
         {                                                                                                            
            "name": "LINKERD2_PROXY_INBOUND_PORTS",
            "value": "9090,9090"                                                                                       
          },
```
then the yaml library, apparently, will wrap that around quotes only if there's a chance of type confusion. In the case of a comma-separated list of numbers, it's clear it's a string, so no quotes are added. In the case of a single number, quotes are added to avoid interpreting the value as a number.

In other golden files in this changeset you'll however see comma-separated-values that still are quoted. Those are from linkerd-control plane's pods, whose yaml is generated simply through the helm templates, and not through json patching, so they're rendered as they're declared in the `_proxy.tpl` file, which is around quotes.

As for dupes, this value comes from a simple string. Detecting dupes would require creating a new partial template with a function that would transform it into a list of ints in order to retrieve uniques and then convert back into a string of comma-separated values. Given that it doesn't make sense to have a pod with duplicate container ports to begin with, I think it's not worth adding that extra complexity.

Reply:
Thanks for digging into this—glad to hear the confusion about how the value gets interpreted is already handled by yaml library we use.

I'm good with not worrying about deduping values either. It's a good point that a pod probably will not have duplicate container ports anyways.